tests/cgroup

```diff
@@ -22,7 +22,7 @@ lxc launch "${IMAGE}" c1
 echo "==> Validate default values"
 [ "$(lxc exec c1 -- nproc)" = "$(nproc)" ]
 [ "$(lxc exec c1 -- grep ^MemTotal /proc/meminfo)" = "$(grep ^MemTotal /proc/meminfo)" ]
-if [ -e "/sys/fs/cgroup/memory/memory.memsw.limit_in_bytes" ] || [ -e "/sys/fs/cgroup/memory.swap.max" ]; then
+if [ -e "/sys/fs/cgroup/memory/memory.memsw.limit_in_bytes" ] || [ -e "/sys/fs/cgroup/system.slice/memory.swap.max" ]; then
```

Comment:
if this is a LXCFS regression, should we rather than updating our tests to avoid the regression, instead cherry-pick the fix into our snap so that the regression doesn't occur. Because if i'm understanding this correctly, this will give the false impression that the tests are passing, which would then lead us to release a snap with a regressed LXCFS?

Reply:
>if this is a LXCFS regression, should we rather than updating our tests to avoid the regression

It is not a LXCFS regression. It was always broken like that with cgroup2.

Reply:
https://github.com/lxc/lxcfs/pull/661 fixes a regression, yes. But it only in the `main` branch and only `latest/edge` LXD snap was affected

https://github.com/lxc/lxcfs/pull/663 just makes swap accounting to work properly with cgroup2. It never worked before.

Reply:
I started to dive into this stuff because of regression (fixed by https://github.com/lxc/lxcfs/pull/661), but then I discovered that our lxd-ci tests didn't work properly on cgroup2 and after I've fixed this I discovered that LXCFS doesn't handle swap accounting on cgroup2 and fixed this by https://github.com/lxc/lxcfs/pull/663. That's a full explanation, I guess.

Reply:
>Because if i'm understanding this correctly, this will give the false impression that the tests are passing, which would then lead us to release a snap with a regressed LXCFS?

no-no, inversely I'm extending our tests to cover cgroup2 scenario. See, we have a checks like:
```
if [ -e "/sys/fs/cgroup/memory/memory.memsw.limit_in_bytes" ] || [ -e "/sys/fs/cgroup/memory.swap.max" ]; then
```

first condition is correct (for cgroup v1), second condition is not correct. It will always fail on the host, because `memory.swap.max` does never exist in a root cgroup. Which means that we don't test cgroup2 stuff at all.

Reply:
I see, thanks for the clarification.